app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupActivity.java

```diff
@@ -73,47 +90,47 @@ protected void disableNextButton() {
      */
     @Override
     public void onClick(View view) {
-        if(mIsNextEnabled && view.getId() == R.id.meetup_next_button){
-            Intent intent = new Intent(this, mMeetup.inReview ? CreateMeetupReviewActivity.class : getNextActivity());
-            //TODO if returning to the review screen, probably remove this screen and the old review screen. Otherwise, back button behavior will be wonky af
-            String json = new Gson().toJson(mMeetup);
-            intent.putExtra(MEETUP_KEY, json);
-            startActivity(intent);
+        if(mIsNextEnabled && (view.getId() == R.id.meetup_next_button || view.getId() == R.id.confirm_button)) {
+            gotoNextActivity(getNextActivity());
         }
+    }
 
+    private void gotoNextActivity(Class nextActivity) {
+        if (isClearingActivity(nextActivity)) {
+            clearPreviousActivities();
+        }
+        Intent intent = new Intent(this, nextActivity);
+        String json = new Gson().toJson(mMeetup);
+        intent.putExtra(MEETUP_KEY, json);
+        startActivity(intent);
     }
 
-    public static final Class[] ACTIVITIES = new Class[]{
-            CreateMeetupLocationActivity.class,
-            CreateMeetupEndTimeActivity.class,
-            CreateMeetupSelectDogsActivity.class,
-            CreateMeetupPhotoActivity.class,
-            CreateMeetupReviewActivity.class
-    };
+    private void clearPreviousActivities() {
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
 
-    public static final Class[] ACTIVITIES_NOSELECTDOG = new Class[]{
-            CreateMeetupLocationActivity.class,
-            CreateMeetupEndTimeActivity.class,
-            CreateMeetupPhotoActivity.class,
-            CreateMeetupReviewActivity.class
-    };
+    private Class getNextActivity() {
+        return getActivityList().get(getNextActivityIndex());
+    }
 
-    public Class getNextActivity() {
-        return getActivityList()[getActivityIndex() + 1];
+    private boolean isClearingActivity(Class activity) {
+        return activity == CreateMeetupReviewActivity.class
```

Comment:
If we're on the Review screen, I don't think pressing back should take us back to the map screen. On the review screen, we should replace the "back" button with a "Cancel" button. When you click "Cancel", a confirmation popup will get rid of it. In this case, the back button would let you go back through the create meetup stack. Otherwise, an accidental back button press wipes out your meetup setup, which could be frustrating, especially with no warning. When a meetup is in progress, that makes sense because you really can't go back to editing it.

Reply:
Sure, I can do that.

Reply:
You don't have to change the button for this PR! Just removing Review from the isClearingActivity is good for now

Reply:
Removed